packages/remote-react-components/dev/vitest/remote-test-server/createTestServer.tsx

```diff
@@ -0,0 +1,36 @@
+import { createServer } from "vite";
+
+const remoteTestServerPort = 6022;
+
+export const createTestServer = () => {
+  const testServer = {
+    start: async () => {
+      const server = await createServer({
+        configFile: "dev/vitest/remote-test-server/vite.config.ts",
+        cacheDir: "node_modules/.vitest-browser",
+        root: "dev/vitest/remote-test-server",
+      });
+
+      testServer.stop = async () => {
+        await server.close();
+        console.log("Remote test server stopped");
+      };
+
+      if (server.httpServer?.address?.["port"] === remoteTestServerPort) {
+        console.error(
+          "Tetst server port is already in use:",
+          remoteTestServerPort,
+        );
+        testServer.stop();
+        return;
+      }
+      await server.listen(remoteTestServerPort);
+      server.printUrls();
+    },
+    stop: async () => {
```

Comment:
it would be more consistent if the server is created beforehand in a variable and start would finally starts listen, so stop can be used as normal with the desired log entry.